huntsman/camera/pyro.py

```diff
@@ -0,0 +1,498 @@
+        """
+        Current status of the camera's image sensor cooling system (enabled/disabled).
+
+        For some cameras it is possible to change this by assigning a boolean
+        """
+        return self._proxy.ccd_cooling_enabled
+
+    @ccd_cooling_enabled.setter
+    def ccd_cooling_enabled(self, enabled):
+        self._proxy.ccd_cooling_enabled = bool(enabled)
+
+    @property
+    def ccd_cooling_power(self):
+        """
+        Current power level of the camera's image sensor cooling system (typically as
+        a percentage of the maximum).
+        """
+        return self._proxy.ccd_cooling_power
+
+# Methods
+
+    def connect(self):
+        """
+        (re)connect to the distributed camera.
+        """
+        self.logger.debug('Connecting to {} on {}'.format(self.name, self._uri))
+
+        # Get a proxy for the camera
+        try:
+            self._proxy = Pyro4.Proxy(self._uri)
+        except Pyro4.errors.NamingError as err:
+            msg = "Couldn't get proxy to camera {}: {}".format(self.name, err)
+            warn(msg)
+            self.logger.error(msg)
+            return
+
+        # Set sync mode
+        Pyro4.asyncproxy(self._proxy, asynchronous=False)
+
+        # Force camera proxy to connect by getting the camera uid.
+        # This will trigger the remote object creation & (re)initialise the camera & focuser,
+        # which can take a long time with real hardware.
+        uid = self._proxy.get_uid()
+        if not uid:
+            msg = "Couldn't connect to {} on {}!".format(self.name, self._uri)
+            warn(msg)
+            self.logger.error(msg)
+            return
+
+        self._connected = True
+        self._serial_number = uid
+        self.model = self._proxy.model
+        self._file_extension = self._proxy.file_extension
+        self._readout_time = self._proxy.readout_time
+        self.filter_type = self._proxy.filter_type
+        self.logger.debug("{} connected".format(self))
+
+    def take_exposure(self,
+                      seconds=1.0 * u.second,
+                      filename=None,
+                      dark=False,
+                      blocking=False,
+                      timeout=None,
+                      *args,
+                      **kwargs):
+        """
+        Take exposure for a given number of seconds and saves to provided filename.
+
+        Args:
+            seconds (u.second, optional): Length of exposure
+            filename (str, optional): Image is saved to this filename
+            dark (bool, optional): Exposure is a dark frame (don't open shutter), default False
+            blocking (bool, optional): If False (default) returns immediately after starting
+                the exposure, if True will block until it completes.
+            timeout (u.second, optional): Length of time beyond the length the exposure to wait
+                for exposures to complete. If not given will wait indefinitely.
```

Comment:
I'm still on the fence about *ever* allowing indefinite waits but it's fine for now. Maybe we need an issue for it.

Reply:
At the least we should be enabling these timeouts in the testing and in the default config files to encourage their use. Will revisit that as part of the general issue of timeouts for all the cameras' threaded actions.